src/Analyzers/Core/Analyzers/PopulateSwitch/PopulateSwitchStatementHelpers.cs

```diff
@@ -133,6 +133,30 @@ private static bool TryRemoveExistingEnumMembers(ISwitchOperation switchStatemen
                         var caseValue = IntegerUtilities.ToInt64(value.ConstantValue.Value);
                         enumValues.Remove(caseValue);
 
+                        break;
+
+                    case CaseKind.Pattern:
+                        foreach (var operation in clause.ChildOperations) // clause is IPatternCaseClauseOperation
+                        {
+                            foreach (var subCase in operation.ChildOperations) // operation is IBinaryPatternOperation
```

Comment:
You need to actually check that you have these types of patterns here.

Reply:
It turns out the switch expression helper already handled this case. Honestly the two helper classes could almost be merged into one but for a few `ISwitchExpressionOperation` vs `ISwitchOperation` differences. I didn't want to make a ton of churn, so I just used the expression helper in the statement helper.